### PR TITLE
Create cdkExecPolicy_tooling.yaml

### PR DESCRIPTION
**Pull request by mourya-33**
CDK bootstrap policies for tooling account

### Feature or Bugfix
- Feature

### Detail
- Cloudformation stack to create least privilege iam policies for bootstrapping CDK to tooling and deployment accounts

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+          - 
+              Sid: VisualEditor10
+              Effect: Allow
+              Action:
+                  - 'cloudformation:DeleteStack'
+                  - 'cloudformation:TagResource'
+                  - 'cloudformation:CreateChangeSet'
+                  - 'cloudformation:UntagResource'
+              Resource:
+                - !Sub 'arn:aws:codeartifact:*:${AwsAccountId}:domain/${EnvironmentResourcePrefix}-*'
```
Comment (noah-paige) [inline deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml:34]: Do we require all these resources for only the set of CloudFormation Actions listed in this statement?

**Review by noah-paige — CHANGES_REQUESTED** [review]
Some Additional Comments:

1. I needed the following additional permissions to have a successful deploy and delete (cross account CICD Deployment):

- [ ] cloudfront:UpdateDistribution
- [ ] ec2:DeleteNetworkAclEntry
- [ ] Ec2:DeleteNetworkAcl
- [ ] ec2:ReplaceNetworkAclAssociation
- [ ] Ec2:CreateNetworkAcl
- [ ] Ec2:CreateNetworkAclEntry
- [ ] Ec2:ReplaceNetworkAclEntry
- [ ] ecr:BatchGetImage
- [ ] ecr:GetDownloadUrlForLayer
- [ ] ECR + CodeArtifact cross account permissions (meaning I edited the resources section of Policy 4 `Sid:VisualEditor8` to include `ecr` and `codeartifact` resources for both the tooling and deployment account)

2.  I think we need to make the parameters more clear - for cross account deployments the way the current params are set up it would not assign the correct permissions

3.I think we should move this as part of the FAQ section in the GitHub Pages on “Deploying to AWS” and have a link there to download the `.yaml` for the execution policies. 

I think since we have not validated that this template works broadly for every configuration type in `cdk.json` and is more to be used as a “getting started” reference asset it makes sense to go in documentation rather than in the main code branch - but curious to hear you’re all thoughts on this @dlpzx @mourya-33

File: deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+              Resource: !Sub 'arn:aws:events:*:${AwsAccountId}:rule/${EnvironmentResourcePrefix}*'
+          -
+              Sid: PassRoleDeploymentAccount
+              Effect: Allow
+              Action: 'iam:PassRole'
+              Resource: !Sub 'arn:aws:iam::${DeploymentAWSAccountId}:role/*'
+          -
+              Effect: Allow
+              Action: 
+                  - 'SNS:GetTopicAttributes'
```
[the remark this answers is not in the document]
Comment (mourya-33) [reply deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml:89]: Done

File: deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+                  - !Sub 'arn:aws:kms:*:${AwsAccountId}:alias/*'
+                  - !Sub 'arn:aws:rds:*:${AwsAccountId}:og:*'
+                  - !Sub 'arn:aws:lambda:*:${AwsAccountId}:*'
+          - 
+              Sid: VisualEditor2
+              Effect: Allow
+              Action:
+                  - 'rds:AddTagsToResource'
+                  - 'rds:DeleteDBSubnetGroup'
+                  - 'lambda:TagResource'
```
[the remark this answers is not in the document]
Comment (mourya-33) [reply deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml:202]: i missed to clean those up. Thank you. Removed them.

File: deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+                  - 'cloudfront:DeleteCloudFrontOriginAccessIdentity'
+                  - 'cloudfront:CreateDistribution'
+                  - 'cloudfront:DeleteDistribution'
+                  - 'cloudfront:GetCloudFrontOriginAccessIdentity'
+                  - 'cloudfront:TagResource'
+                  - 'cloudfront:UntagResource'
+                  - 'cloudfront:GetDistribution'
+              Resource:
+                  - !Sub 'arn:aws:cloudfront::${AwsAccountId}:origin-access-identity/*'
+                  - !Sub 'arn:aws:cloudfront::${AwsAccountId}:distribution/*'
```
[the remark this answers is not in the document]
Comment (mourya-33) [reply deploy/cdk_exec_policy_tooling/cdkExecPolicy_tooling.yaml]: Done

**Comment by dlpzx** [comment]
@mourya-33 this PR needs to be added to `v2m1m0` branch right? to be merged alongside the one with the commands to deploy it #819

**Comment by anmolsgandhi** [comment]
@dlpzx This is pointed at modularization-main, do we need this PR or it is outdated now ?

**Comment by noah-paige** [comment]
@mourya-33 - what is the status of this PR? There are still some pending comments above and we would need to re-test the execution policy for the latest set of features that can data.all support before we can review and merge this

Depending on the priority of this PR we can also close this for now and re-visit when some additional bandwidth is available - do let me know your thoughts?

cc: @anmolsgandhi @dlpzx

**Comment by noah-paige** [comment]
Due to this PRs inactivity (last update on Aug 11) we will be closing it for now. Please re-open if it is required and additional changes are addressed, thanks

**Pull request closed without merging**